collector/prometheus.go

```diff
@@ -65,113 +58,140 @@ func getBearerToken(tokenFile string) (config.Secret, error) {
 	return config.Secret(encodedSecret), nil
 }
 
-func getPrometheusConfig(kmCfg *kokumetricscfgv1beta1.PrometheusSpec, inCluster bool) (*PrometheusConfig, error) {
-	if !inCluster {
-		val, ok := os.LookupEnv("SECRET_ABSPATH")
-		if ok {
-			serviceaccountPath = val
+func statusHelper(cr *kokumetricscfgv1beta1.KokuMetricsConfig, status string, err error) {
+	switch status {
+	case "configuration":
+		if err != nil {
+			cr.Status.Prometheus.PrometheusConfigured = false
+			cr.Status.Prometheus.ConfigError = fmt.Sprintf("%v", err)
+		} else {
+			cr.Status.Prometheus.PrometheusConfigured = true
+			cr.Status.Prometheus.ConfigError = ""
+		}
+	case "connection":
+		if err != nil {
+			cr.Status.Prometheus.PrometheusConnected = false
+			cr.Status.Prometheus.ConnectionError = fmt.Sprintf("%v", err)
+		} else {
+			cr.Status.Prometheus.PrometheusConnected = true
+			cr.Status.Prometheus.ConnectionError = ""
 		}
 	}
-	promCfg := &PrometheusConfig{
-		Address: kmCfg.SvcAddress,
-		CAFile:  filepath.Join(serviceaccountPath, certKey),
-		SkipTLS: *kmCfg.SkipTLSVerification,
+}
+
+type PrometheusConfigurationSetter func(ps *kokumetricscfgv1beta1.PrometheusSpec, c *PrometheusCollector) error
+
+func SetPrometheusConfig(ps *kokumetricscfgv1beta1.PrometheusSpec, c *PrometheusCollector) error {
```

Comment:
Short variable names hinder readability. Even though this function is pretty short, it would still be better to have more descriptive variable and parameter names.
```suggestion
func SetPrometheusConfig(promspec *kokumetricscfgv1beta1.PrometheusSpec, collector *PrometheusCollector) error {
```

Reply:
I understand that perspective but `c` for `PrometheusCollector` is and has been used everywhere else. `ps` was `kmCfg` which made no sense. If there really is an issue with readability, we can address that in a follow up PR.

Reply:
Sounds good.